bcdaworker/main.go

```diff
@@ -204,9 +204,24 @@ func writeBBDataToFile(bb client.APIClient, acoID string, acoCMSID string, cclfB
 	totalBeneIDs := float64(len(cclfBeneficiaryIDs))
 	failThreshold := getFailureThreshold()
 	failed := false
+	suppressedList := models.GetSuppressedBlueButtonIDs()
 
 	for _, cclfBeneficiaryID := range cclfBeneficiaryIDs {
 		blueButtonID, err := beneBBID(cclfBeneficiaryID, bb)
+		foundSuppressed := false
+
+		for _, suppressedBBID := range suppressedList {
```

Comment:
Instead of iterating, maybe just a query is all we need; each query should be fast, due to the indexes.  Something like below (it might need a little work, but this is the general idea):

```
SELECT blue_button_id
FROM suppressions
WHERE blue_button_id=? AND effective_date <= NOW() AND preference_indicator != '';
```

The `?` above would be the blueButtonID returned from `beneBBID`.

If we have a match, then we skip this bene (since the bene is suppressed).  Otherwise, we continue.